[Issue]: cmake_minimum_required is too old

The top CMakeLists.txt sets cmake_minimum_required to 3.5.0

https://github.com/ROCm/ROCmValidationSuite/blob/bd1954192c91a372a2afed92105b6142fbd2b46e/CMakeLists.txt#L26

but some components still set the version to 2.x:

https://github.com/ROCm/ROCmValidationSuite/blob/bd1954192c91a372a2afed92105b6142fbd2b46e/CMakeGtestDownload.cmake#L26

Such files are configured but CMAKE_POLICY_VERSION_MINIMUM is not propagated

https://github.com/ROCm/ROCmValidationSuite/blob/bd1954192c91a372a2afed92105b6142fbd2b46e/CMakeLists.txt#L423-L430

Consequently configuration fails (cmake 4.2.3):

```
CMake Error at CMakeLists.txt:26 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

--- a/CMakeGtestDownload.cmake
+++ b/CMakeGtestDownload.cmake
@@ -23,7 +23,7 @@
 ##
 ################################################################################
 
-cmake_minimum_required(VERSION 2.8.2)
+cmake_minimum_required(VERSION 3.5.0)
 
 project(googletest-download NONE)
 

--- a/CMakeLists.txt.yaml
+++ b/CMakeLists.txt.yaml
@@ -23,7 +23,7 @@
 ##
 ################################################################################
 
-cmake_minimum_required(VERSION 2.8.2)
+cmake_minimum_required(VERSION 3.5.0)
 
 project(yaml-download NONE)
 

--- a/CMakeMXDataGeneratorDownload.cmake
+++ b/CMakeMXDataGeneratorDownload.cmake
@@ -23,7 +23,7 @@
 ##
 ################################################################################
 
-cmake_minimum_required(VERSION 2.8.2)
+cmake_minimum_required(VERSION 3.5.0)
 
 project(mxDataGenerator-download NONE)
 

--- a/CMakeRBLASDownload.cmake
+++ b/CMakeRBLASDownload.cmake
@@ -23,7 +23,7 @@
 ##
 ################################################################################
 
-cmake_minimum_required(VERSION 2.8.2)
+cmake_minimum_required(VERSION 3.5.0)
 
 project(rvs_rblas-download NONE)
 

--- a/CMakeRSMIDownload.cmake
+++ b/CMakeRSMIDownload.cmake
@@ -23,7 +23,7 @@
 ##
 ################################################################################
 
-cmake_minimum_required(VERSION 2.8.2)
+cmake_minimum_required(VERSION 3.5.0)
 
 project(rvs_smi-download NONE)
 

